lib/jekyll/filters/url_filters.rb

```diff
@@ -29,6 +29,23 @@ def relative_url(input)
         ).normalize.to_s
       end
 
+      # Produces the path to the given asset relative to the assets directory at the
+      # domain root based on site.baseurl
+      #
+      # input - the asset file relative to the 'assets' directory. The string may also
+      #         include any subfolder(s) the asset may be a part of.
+      #           e.g. "main.css" or "css/main.css"
+      #
+      # Returns a path relative to the assets directory as a String.
+      def asset_url(input)
+        return if input.nil?
```

Comment:
If I'm honest, I think filters should just throw instead of returning unexpected results if the user makes a mistake.  AKA if I use the wrong variable name, or something, this will give an expected result when it should have thrown.